<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog_router</class>
 <widget class="QDialog" name="Dialog_router">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>809</width>
    <height>556</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Router</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>775</width>
        <height>602</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Unique description of a specific router.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>router_id</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_router_id">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Optional name of catalog where a target table is located. If this field is unspecified, the catalog will be either the default catalog at the target node or the &amp;quot;source catalog name&amp;quot; from the table trigger, depending on how &amp;quot;use source catalog schema&amp;quot; is set for the router. Variables are substituted for &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(sourceNodeId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(sourceExternalId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(sourceNodeGroupId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(targetNodeId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(targetExternalId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(targetNodeGroupId)&lt;/span&gt;, and &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(none)&lt;/span&gt;. Parameter values can be substituted using &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(name)&lt;/span&gt; syntax.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>target_catalog_name</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_target_catalog_name">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Optional name of schema where a target table is located. If this field is unspecified, the schema will be either the default schema at the target node or the &amp;quot;source schema name&amp;quot; from the table trigger, depending on how &amp;quot;use source catalog schema&amp;quot; is set for the router. Variables are substituted for &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(sourceNodeId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(sourceExternalId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(sourceNodeGroupId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(targetNodeId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(targetExternalId)&lt;/span&gt;, &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(targetNodeGroupId)&lt;/span&gt;, and &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(none)&lt;/span&gt;. Parameter values can be substituted using &lt;span style=&quot; font-family:'monospace';&quot;&gt;$(name)&lt;/span&gt; syntax.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>target_schema_name</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_target_schema_name">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Optional name for a target table. Only use this if the target table name is different than the source.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>target_table_name</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_target_table_name">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>source_node_group_id</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_source_node_group_id">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>target_node_group_id</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_target_node_group_id">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_14">
         <item>
          <widget class="QLabel" name="label_15">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The type of router. Standard router types are listed below. Custom routers can be configured as extension points. &lt;/p&gt;&lt;p&gt;Table 2. Router Types &lt;/p&gt;&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;&lt;tr&gt;&lt;td&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Type&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Provided&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Description&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;default&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;x&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;A router that sends all captured data to all nodes that belong to the target node group defined in the router. See &lt;a href=&quot;#_default_router&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;Default Router&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;column&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;x&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;A router that compares old or new column values in a captured data row to a constant value or the value of a target node’s external id or node id. See &lt;a href=&quot;#_column_match_router&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;Column Match Router&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;audit&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;x&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;A router that inserts into an automatically created audit table. It records captured changes to tables that it is linked to. See &lt;a href=&quot;#_audit_table_router&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;Audit Table Router&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;java&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;x&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;A router that executes a Java expression in order to select nodes to route to. The script can use the old and new column values. See &lt;a href=&quot;#Java Router&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;[Java Router]&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;lookuptable&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;x&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;A router which can be configured to determine routing based on an existing or ancillary table specifically for the purpose of routing data. See &lt;a href=&quot;#_lookup_table_router&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;Lookup Table Router&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;subselect&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;x&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;A router that executes a SQL expression against the database to select nodes to route to. This SQL expression can be passed values of old and new column values. See &lt;a href=&quot;#_subselect_router&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;Subselect Router&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;bsh&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;x&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;A router that executes a Bean Shell script expression in order to select nodes to route to. The script can use the old and new column values. See &lt;a href=&quot;#_beanshell_router&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;Beanshell Router&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;csv&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;x&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;A router that sends data from a CSV file to a target table of the target node group defined in the router. See &lt;a href=&quot;#_csv_router&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;CSV Router&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;dbf&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;x&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;A router that sends data from a dBase generated DBF file to a target table of the target node group defined in the router. See &lt;a href=&quot;#_dbf_router&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;DBF Router&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>router_type</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_router_type">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>default</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>column</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>audit</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>java</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>lookuptable</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>subselect</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>bsh</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>csv</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>dbf</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>router_expression</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="plainTextEdit_router_expression">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>150</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QLabel" name="label_11">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Flag that indicates that this router should send updated rows from nodes in the source node group to nodes in the target node group.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>sync_on_update</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_sync_on_update">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_11">
         <item>
          <widget class="QLabel" name="label_12">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Flag that indicates that this router should send inserted rows from nodes in the source node group to nodes in the target node group.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>sync_on_insert</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_sync_on_insert">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_13">
         <item>
          <widget class="QLabel" name="label_14">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Flag that indicates that this router should send deleted rows from nodes in the source node group to nodes in the target node group.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>sync_on_delete</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_sync_on_delete">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_12">
         <item>
          <widget class="QLabel" name="label_13">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;If set then the source catalog and source schema are sent to the target to be used to find the target table.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>use_source_catalog_schema</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_use_source_catalog_schema">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_16">
         <item>
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>description</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_description">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>28</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_19">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="minimumSize">
        <size>
         <width>211</width>
         <height>41</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>211</width>
         <height>41</height>
        </size>
       </property>
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
